codec-http/src/main/java/io/netty/handler/codec/http/HttpObjectDecoder.java

```diff
@@ -600,23 +600,62 @@ private State readHeaders(ByteBuf buffer) {
         if (name != null) {
             headers.add(name, value);
         }
+
         // reset name and value fields
         name = null;
         value = null;
 
-        State nextState;
+        List<String> values = headers.getAll(HttpHeaderNames.CONTENT_LENGTH);
+        int size = values.size();
+        if (size > 0) {
+            contentLength = Long.parseLong(values.get(0));
```

Comment:
nit: should this be done after the `if` check below to avoid parsing the `content-length` header if there are multiple such headers?:

```java
if (message.protocolVersion() == HttpVersion.HTTP_1_1) {
 .....
}
contentLength = Long.parseLong(values.get(0));
```